client/host_queue.go

```diff
@@ -192,37 +190,36 @@ func (q *queue) drain() {
 		writeBatchSize               = q.opts.WriteBatchSize()
 	)
 
-	for {
-		ops, ok := <-q.drainIn
-		if !ok {
-			break
-		}
+	for ops := range q.drainIn {
+		ops := ops
 		var (
 			currWriteOps      []op
 			currBatchElements []*rpc.WriteBatchRawRequestElement
 			opsLen            = len(ops)
 		)
+
 		for i := 0; i < opsLen; i++ {
 			switch v := ops[i].(type) {
 			case *writeOp:
 				namespace := v.namespace
-				currWriteOps = currWriteOpsByNamespace[namespace.Hash()]
-				currBatchElements = currBatchElementsByNamespace[namespace.Hash()]
+				hash := namespace.Hash()
+				currWriteOps = currWriteOpsByNamespace[hash]
+				currBatchElements = currBatchElementsByNamespace[hash]
 				if currWriteOps == nil {
 					currWriteOps = q.opsArrayPool.Get()
 					currBatchElements = q.writeBatchRawRequestElementArrayPool.Get()
 				}
 
 				currWriteOps = append(currWriteOps, ops[i])
 				currBatchElements = append(currBatchElements, &v.request)
-				currWriteOpsByNamespace[namespace.Hash()] = currWriteOps
-				currBatchElementsByNamespace[namespace.Hash()] = currBatchElements
+				currWriteOpsByNamespace[hash] = currWriteOps
+				currBatchElementsByNamespace[hash] = currBatchElements
 
 				if len(currWriteOps) == writeBatchSize {
 					// Reached write batch limit, write async and reset
 					q.asyncWrite(wgAll, namespace, currWriteOps, currBatchElements)
-					currWriteOpsByNamespace[namespace.Hash()] = nil
-					currBatchElementsByNamespace[namespace.Hash()] = nil
+					currWriteOpsByNamespace[hash] = nil
```

Comment:
This and in other places should probably be delete(currWriteOpsByNamespace, hash), I think?